client/src/featureform/register.py

```diff
@@ -2415,9 +2440,9 @@ def apply(self):
             return
 
         if self.local:
-            state().create_all_local()
+            self.state().create_all_local()
         else:
-            state().create_all(self._stub)
+            self.state().create_all(self._stub)
```

Comment:
@sdreyer, this is the issue that caused me to chase my own tail for a bit. I've run the tests and it doesn't appear to impact the hosted mode, but let me know what you think about this change.

Reply:
interesting, def looks good to me

Reply:
@sdreyer, I think the issue was that even when I wasn't using the global `ff` (i.e. the global registrar) my resources were being applied to it because the code path was calling the global `state` method even when I had a "local" resource client. When I traced down the issue, I saw this and figured it should be `self.state` in both cases given that would still allow the global registrar to behave as it has been, but a local instance of the resource client would hold its own registrar. Does this make sense?